--- vtkInteractorStyleSwitch.cxx.old	2005-01-11 16:29:41.000000000 +0100
+++ vtkInteractorStyleSwitch.cxx	2005-02-21 09:58:29.000000000 +0100
@@ -143,6 +143,8 @@
 //----------------------------------------------------------------------------
 // this will do nothing if the CurrentStyle matchs
 // JoystickOrTrackball and CameraOrActor
+// It should! If the this->Interactor was changed (using SetInteractor()),
+// and the currentstyle should not change.
 void vtkInteractorStyleSwitch::SetCurrentStyle()
 {
   // if the currentstyle does not match JoystickOrTrackball 
@@ -161,7 +163,6 @@
         this->CurrentStyle->SetInteractor(0);
         }
       this->CurrentStyle = this->JoystickCamera;
-      this->CurrentStyle->SetInteractor(this->Interactor);
       }
     }
   else if (this->JoystickOrTrackball == VTKIS_JOYSTICK &&
@@ -174,7 +175,6 @@
         this->CurrentStyle->SetInteractor(0);
         }
       this->CurrentStyle = this->JoystickActor;
-      this->CurrentStyle->SetInteractor(this->Interactor);
       }
     }
   else if (this->JoystickOrTrackball == VTKIS_TRACKBALL &&
@@ -187,7 +187,6 @@
         this->CurrentStyle->SetInteractor(0);
         }
       this->CurrentStyle = this->TrackballCamera;
-      this->CurrentStyle->SetInteractor(this->Interactor);
       }
     }
   else if (this->JoystickOrTrackball == VTKIS_TRACKBALL &&
@@ -200,9 +199,12 @@
           this->CurrentStyle->SetInteractor(0);
           }
         this->CurrentStyle = this->TrackballActor;
-        this->CurrentStyle->SetInteractor(this->Interactor);
         }
     }
+  if (this->CurrentStyle)
+    {
+    this->CurrentStyle->SetInteractor(this->Interactor);
+    }
 }
 
 //----------------------------------------------------------------------------
@@ -245,3 +247,22 @@
     }
 }
 
+//----------------------------------------------------------------------------
+void vtkInteractorStyleSwitch::SetDefaultRenderer(vtkRenderer* renderer)
+{
+  this->vtkInteractorStyle::SetDefaultRenderer(renderer);
+  this->JoystickActor->SetDefaultRenderer(renderer);
+  this->JoystickCamera->SetDefaultRenderer(renderer);
+  this->TrackballActor->SetDefaultRenderer(renderer);
+  this->TrackballCamera->SetDefaultRenderer(renderer);
+}
+
+//----------------------------------------------------------------------------
+void vtkInteractorStyleSwitch::SetCurrentRenderer(vtkRenderer* renderer)
+{
+  this->vtkInteractorStyle::SetCurrentRenderer(renderer);
+  this->JoystickActor->SetCurrentRenderer(renderer);
+  this->JoystickCamera->SetCurrentRenderer(renderer);
+  this->TrackballActor->SetCurrentRenderer(renderer);
+  this->TrackballCamera->SetCurrentRenderer(renderer);
+}
