--- vtkInteractorStyleSwitch.h.old	2003-11-14 21:43:37.000000000 +0100
+++ vtkInteractorStyleSwitch.h	2005-02-21 09:58:29.000000000 +0100
@@ -67,6 +67,12 @@
   // Only care about the char event, which is used to switch between
   // different styles.
   virtual void OnChar();
+
+  // Description:
+  // Overridden from vtkInteractorObserver because the interactor styles
+  // used by this class must also be updated.
+  virtual void SetDefaultRenderer(vtkRenderer*);
+  virtual void SetCurrentRenderer(vtkRenderer*);
   
 protected:
   vtkInteractorStyleSwitch();
